test/format/ops.c

```diff
@@ -64,6 +64,39 @@ modify_repl_init(void)
 		modify_repl[i] = "zyxwvutsrqponmlkjihgfedcba"[i % 26];
 }
 
+/*
+ * on_alarm --
+ *	Handle alarm signal.
+ */
+static void on_alarm(int signo) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+static void
+on_alarm(int signo)
+{
+	(void)signo;                            /* Unused parameter */
+
+	fprintf(stderr, "%s\n",
+	    "format run attempted to dump transaction and cache state, "
+	    "but then timed out, aborting the process.");
+	__wt_abort(NULL);
+}
+
+static void
+set_alarm(void)
+{
+#ifndef _WIN32
+	struct itimerspec timer_val;
+	timer_t timer_id;
+
+	(void)signal(SIGALRM, on_alarm);
+
+	testutil_check(timer_create(CLOCK_REALTIME, NULL, &timer_id));
+	memset(&timer_val, 0, sizeof(timer_val));
+	timer_val.it_value.tv_sec = 60 * 2;
```

Comment:
I think one minute or even 30 seconds is sufficient. Two minutes seems like overkill.

Reply:
I agree, but I want to avoid any chance of a false positive, that is, I don't want any risk of debugging cases where the timer went off erroneously (and I didn't see any advantage in tightening down the timer). If you can think of a reason to tighten down the timer, let me know,